packages/fetch-http-handler/src/stream-collector.ts

```diff
@@ -2,13 +2,17 @@ import { StreamCollector } from "@aws-sdk/types";
 import { fromBase64 } from "@aws-sdk/util-base64-browser";
 
 //reference: https://snack.expo.io/r1JCSWRGU
-export const streamCollector: StreamCollector = (stream: Blob | ReadableStream): Promise<Uint8Array> => {
-  if (stream instanceof Blob) {
-    return collectBlob(stream);
-  }
+export const streamCollector: StreamCollector =
+  // `Blob` does not exist on Cloudflare Workers.
+  typeof Blob === "undefined"
```

Comment:
If I understand the purpose of this code correctly, this could even be a feature check on `response.body` instead so `collectBlob` or `collectStream` would only run when the response follows the same logic in https://github.com/aws/aws-sdk-js-v3/blob/master/packages/fetch-http-handler/src/fetch-http-handler.ts#L74

Reply:
Actually, now that I think about this IE doesn’t even support `fetch`. Is there a specific reason for using `blob()`?

Reply:
Potential feature check that would cover both:

```
'body' in Response.prototype
```

This would allow all non-IE browsers to run using the optimized stream code path while IE runs in the blob path, and then there's no need for checking `stream instanceof Blob` anymore.

Reply:
To answer your question:
> Actually, now that I think about this IE doesn’t even support fetch. Is there a specific reason for using blob()?

This is necessary for old browsers without Stream API support, and React-Native. 

This code would look nicer if

```javascript
export const streamCollector: StreamCollector = (stream: Blob | ReadableStream): Promise<Uint8Array> => {
  if (typeof Blob === "function" &&  stream instanceof Blob) {
    return collectBlob(stream);
  }
  return collectStream(stream);
}
```

Reply:
Wouldn't `'body' in Response.prototype` handle this? If it does, you could do away with the runtime check altogether. I don't have an old browser or React Native set up to test, but reading over the `fetch` API by GitHub it would return `false` while modern browsers return `true`.

Reply:
Of course, happy to do it in the runtime code path if you do prefer, it's not a big difference.

Reply:
Also, using `typeof Blob === "function" &&  stream instanceof Blob` won't pass the type checker without type assertions. Is that ok?

Reply:
Regarding using `'body' in Response.prototype` in the Fetch HTTP handler, I prefer to have the feature check in stream collector too. Because stream collector and Fetch http handler can be exported separately, I don't like them to be intertwined too much.